suite/tests/CMakeLists.txt

```diff
@@ -2624,23 +2624,35 @@ if (X86)
       set(client.drx-scattergather_runavx 1)
     endif ()
 
-    if (proc_supports_avx AND X64)
-      add_exe(allasm_scattergather
-        ${PROJECT_SOURCE_DIR}/clients/drcachesim/tests/allasm_scattergather.asm
+    if (X64 AND UNIX)
+      if (proc_supports_avx)
+        add_exe(allasm_scattergather
+          ${PROJECT_SOURCE_DIR}/clients/drcachesim/tests/allasm_scattergather.asm
+          "-early_inject" "")
+        set_target_properties(allasm_scattergather PROPERTIES LINKER_LANGUAGE C)
+        append_link_flags(allasm_scattergather
+          "-nostartfiles -nodefaultlibs -static")
+        append_property_string(TARGET allasm_scattergather LINK_FLAGS
+          # The default value of CMAKE_SHARED_LIBRARY_LINK_C_FLAGS has -rdynamic, which
+          # with some tool chains (such as SUSE GCC 6.2) results in an executable that
+          # is not properly static. Adding --no-export-dynamic avoids the problem.
```

Comment:
How about a reference rather than full 3-line copy-and-paste; or a function/macro to share this common sequence.

Reply:
Done.

Reply:
Added a macro.